src/components/reference/reference.ts

```diff
@@ -0,0 +1,19 @@
+import './reference.scss';
+import WebComponent, { createElementFromString } from '../../common/WebComponent';
+import Router from '../../common/Router';
+
+class ReferenceComponent implements WebComponent {
+  constructor(private link: string, private name: string) {}
+
+  render(): Element {
+    const router = new Router(document.querySelector('#root')!);
+    return createElementFromString(`<a href="#/${this.link}" 
```

Comment:
I cannot go to the root site using this component. If the link is an empty string then it's not working.
You need to change it so if link is an empty string `''` then there is no forward slash in `href`, just `#`.

Or you could do even better and make the `link` argument optional, using `?` sign.
`constructor(private name: string, private link?: string) {}`

Reply:
I had a problem with the optional parameter because it is returned as `undefined` so I changed it that way and it should be working now.